snippets/fsharp/VS_Snippets_CLR_System/system.console.class/fs/example3.fs

```diff
@@ -0,0 +1,169 @@
+﻿
+// dump a range of Unicode characters as a 16x16 array
+// <Snippet4>
+open System
+open System.IO
+open System.Globalization
+open System.Text
+
+type uint = uint32
+
+let roundDownToMultipleOf (b:uint) (u:uint) : uint =
+    u - (u % b)
+
+let roundUpToMultipleOf (b:uint) (u:uint) : uint =
+    roundDownToMultipleOf b u |> (+) b
+
+let displayRange (start:uint) (``end``:uint) : unit =
+
+    let mutable upperRange : uint = 0x10FFFFu
+    let mutable surrogateStart : uint = 0xD800u
+    let mutable surrogateEnd : uint = 0xDFFFu
+   
+    let mutable start = start
+    let mutable ``end`` = ``end``
+    if ``end`` <= start
+    then
+        let t = start
+        start <- ``end``
+        ``end`` <- t
```

Comment:
This kind of pattern is usually done via shadowing and a tuple-returning expression:

```fsharp
    let start, ``end`` =
        if ``end`` <= start then
            ``end``, start
        else 
            start, ``end``
```

These will shadow the input parameters now that the names have been rebound.

Reply:
Good to know. I think my original code was shadowing just the same but didn't not destructure the tuple like in your example. I like avoiding `mutable` as much as possible though so I'll adopt your solution.